src/renderer/flux/Store.ts

```diff
@@ -45,48 +45,38 @@ export default class Store<T> {
   }
 
   getNextState(type: string, payload: any): T {
-    let nextState: T = this.reduce(type, payload);
-
-    if (nextState === undefined) {
-      throw new Error(`${this.constructor.name}.reduce returns undefined, action type: ${type}`);
-    }
-
-    if (nextState instanceof StateBuilder) {
-      nextState = nextState.end();
-    }
-
-    return nextState;
+    return this.reduce(type, payload).end();
```

Comment:
TypeScript guarantees all `Store` classes must define `reduce` does not returns undefined.